src/idlcxx/src/streamers.c

```diff
@@ -1129,20 +1116,36 @@ print_switchbox_open(struct streams *streams)
 
 static idl_retcode_t
 print_constructed_type_close(
-  struct streams *streams)
+  struct streams *streams,
+  const void* node)
 {
   const char *fmt =
-    "  return streamer.finish_struct(*props);\n"
+    "  return streamer.finish_struct(*props, member_ids);\n"
     "}\n\n";
   static const char *pfmt =
     "\n  entity_properties_t::finish(props, keylist);\n"
-    "  props_end = props.data() + props.size();\n"
     "  initialized.store(true, std::memory_order_release);\n"
     "  return props;\n"
     "}\n\n";
+  static const char *pfmt2 =
+    "static const propvec &properties_%1$s = get_type_props<%2$s>();\n\n";
```

Comment:
If this `static const` variable is not used anywhere, the compiler can optimize this and could lead to the same lifetime issues, correct?

Reply:
> If this `static const` variable is not used anywhere, the compiler can optimize this and could lead to the same lifetime issues, correct?

Pulling the following simple example program through godbolt seems to call the "initialization function" `f()` during the global initialization step for the major compilers (GCC x86-64, CLang x86-64, MSVC x64) even at the highest optimization level (-O3):
```
const int fac(int i);

const int fac(int i)
{
if (i)
    return i*fac(i-1);
else
    return 1;
}

const int& f()
{
static int m_i = fac(4);
return m_i;
}

const int &ref_i = f();
 
int main()
{
    return f();
}
```

So I think your fears regarding the function call being optimized away are unfounded.
I could be wrong, as this is very much outside my zone of experience (and comfort) regarding C++ :)

Reply:
@reicheratwork Shouldn't be `static const int &ref_i = f()`, to be exact for our scenario?

I have slightly simplified your example and check this godbolt link https://godbolt.org/z/nbhc5h5xM if we don't have the attribute `__attribute__((used))` or `volatile`, the unused static const is actually optimized out when compiled, is my understanding incorrect here?

Reply:
@sumanth-nirmal  I think you are correct in this regard, I will add the additional change to the PR.

Reply:
Looks like this does not work for Windows, let me look further.

Reply:
@reicheratwork Windows is definitely out of my league, but GNU `_attribute__` equivalent  in MSVC is `__declspec` but not sure if it supports an attribute similar to what we are need here

Reply:
> @reicheratwork Windows is definitely out of my league, but GNU `_attribute__` equivalent in MSVC is `__declspec` but not sure if it supports an attribute similar to what we need here

Exactly, it looks like MSVC does not have a "used" attribute or something equivalent.

Reply:
@reicheratwork Looks good to me.

We just a need a follow-up issues for fixing or agreeing on the limitation with the lifetime of the entities